plugins/tailor.cpp

```diff
@@ -169,50 +171,34 @@ class Tailor {
     {
         available.clear();
         needed.clear();
-        queued.clear();
         sizes.clear();
-        orders.clear();
         supply.clear();
+        orders.clear();
     }
 
     void scan_clothing()
     {
-        if (!inventory_sanity_checking)
+        for (auto i : world->items.other[df::items_other_id::ANY_GENERIC37]) // GENERIC37 is "nontattered clothing"
         {
-            for (auto i : world->items.other[df::items_other_id::ANY_GENERIC37]) // GENERIC37 is "clothing"
+            if (i->flags.whole & badFlags.whole)
             {
-                if (i->flags.whole & badFlags.whole)
-                    continue;
-                if (i->getWear() >= 1)
-                    continue;
-                df::item_type t = i->getType();
-                int size = world->raws.creatures.all[i->getMakerRace()]->adultsize;
-
-                available[std::make_pair(t, size)] += 1;
+                continue;
             }
-        }
-        else
-        {
-            auto& l = world->items.other[df::items_other_id::ANY_GENERIC37];
-
-            for (auto i : world->items.other[df::items_other_id::IN_PLAY])
-            {
-                if (i->flags.whole & badFlags.whole)
-                    continue;
-                if (!i->isClothing())
-                    continue;
-                if (std::find(std::begin(l), std::end(l), i) == std::end(l))
-                {
-                    DEBUG(cycle).print("tailor: clothing item %d missing from GENERIC37 list\n", i->id);
-                }
-                if (i->getWear() >= 1)
-                    continue;
-                df::item_type t = i->getType();
-                int size = world->raws.creatures.all[i->getMakerRace()]->adultsize;
+            if (i->getWear() >= 1)
+                continue;
+            df::item_type t = i->getType();
+            int size = world->raws.creatures.all[i->getMakerRace()]->adultsize;
 
-                available[std::make_pair(t, size)] += 1;
-            }
+            available[std::make_pair(t, size)] += 1;
+        }
 
+        for (auto& i : available)
```

Comment:
if this loop is just for debugging, you can enclose it in an if block that tests whether the output will go anywhere:
```cpp
if (DBG_NAME(cycle).isEnabled(DebugCategory::LDEBUG))
```